config/src/main/java/com/yahoo/vespa/config/JRTConnectionPool.java

```diff
@@ -66,19 +66,31 @@ public synchronized JRTConnection getCurrent() {
     }
 
     @Override
-    public synchronized JRTConnection switchConnection() {
+    public synchronized JRTConnection switchConnection(Connection failingConnection) {
```

Comment:
An alternative strategy here may be to:
* Return current connection if unequal to failingConnection, or
* return a random connection != failingConnection.

There would then be no need for isHealthy, or the 3 `last` timestamps.

Reply:
Good suggestion, will implement